builtin/tag.c

```diff
@@ -106,7 +106,28 @@ static int delete_tag(const char *name, const char *ref,
 static int verify_tag(const char *name, const char *ref,
 				const unsigned char *sha1)
 {
-	return gpg_verify_tag(sha1, name, GPG_VERIFY_VERBOSE);
+	int flags;
+	flags = GPG_VERIFY_VERBOSE;
+
+	if (fmt_pretty) {
+		//verify_ref_format(fmt_pretty);
+
+		// Create a `ref_array_item` in order to use `show_ref_array_item`
+		// XXX: Just copies code from `static new_ref_array_item` in ref_filter.c
+		struct ref_array_item *ref_item;
+		FLEX_ALLOC_STR(ref_item, refname, name);
+		hashcpy(ref_item->objectname, sha1);
+		ref_item->kind = FILTER_REFS_TAGS;
+		ref_item->flag = 0;
+
+		show_ref_array_item(ref_item, fmt_pretty, 0);
+		free((char *)ref_item->symref);
+		free(ref_item);
+
```

Comment:
The allocation, initialisation and deallocation in [lines 117-125](https://github.com/lukpueh/git/blob/010fc57206b74febc2b4bc9ad21510e26950b406/builtin/tag.c#L117-L125) actually just copy the code from [`new_ref_array_item`](https://github.com/lukpueh/git/blob/master/ref-filter.c#L1320-L1330) and [`free_array_item`](https://github.com/lukpueh/git/blob/master/ref-filter.c#L1423-L1427) respectively, which are currently restricted to the scope of [`ref-filter.c`](https://github.com/lukpueh/git/blob/master/ref-filter.c). 

I suggest to add another patch that makes those functions publicly available and also proposes a name refactor. There is no need to *bind* [`struct ref_array_item`](https://github.com/lukpueh/git/blob/master/ref-filter.h#L35-L43) and it's related functions (new_ref_array_item, show_ref_array_item, free_array_item) to [`struct ref_array`](https://github.com/lukpueh/git/blob/master/ref-filter.h#L45-L49) by their names.